### ISSUE TITLE
Think about a configuration option for an "own org"

### ISSUE THREAD

**Issue by sschuberth** [oss-review-toolkit/ort#4892]
What do you guys think about introducing a configuration option as a list of strings that define what's to be considered the own organization in terms of `Identifier.isFromOrg()`? Then we could introduce an `isFromOwnOrg()` helper for rules to handle things differently when dealing with own packages.

**Comment by mnonnenmacher** [comment]
That's already trivial to implement with:
```kotlin
fun isFromOwnOrg() = isFromOrg("some", "custom", "org", "strings")
```

`isFromOrg` also needs some improvements, it only supports Gradle, Maven, NPM, and SBT, but we set a namespace in all of these: Carthage, Composer, Gradle, Maven, NPM, Pub, SBT, SpdxDocumentFile, Stack, Yarn

I'm also not sure how well the heuristic in `isFromOrg` works for random organizations. Maybe replacing it with a list of regular expressions that are matched against the identifier would be a better option. Then you could define patterns like `Maven:com\.example\..*` or `PIP::myorg-.*`.

**Comment by sschuberth** [comment]
> That's already trivial to implement with

Right, but my main question was whether we should move the `"some", "custom", "org", "strings"` part to a dedicated `ort.conf` option, or whether that's overkill. I.e. do we want to make the concept of an "own org" a "first-class citizen"?

**Comment by mnonnenmacher** [comment]
For this single use case it seems overkill to me, but I'm also not strictly against it. I brought up the other points because I thought if this configuration should be reusable in other places we might need a more sophisticated approach.
